### PR TITLE
Calico metrics scrape

### LINKED ISSUE #527: [] Enable calico metrics scrape
**Is your feature request related to a problem? Please describe.**
Calico pods metrics are not scraped by Prometheus.

**Describe the solution you'd like**
I want Prometheus to scrape calico pods for metrics.

**Describe alternatives you've considered**
none

**Additional context**
node

**Definition of done:**
When calico pods are discovered and scraped by Prometheus.

### PR DESCRIPTION
**Pull request by crssnd**
**What this PR does / why we need it**: this will allow us to scrape calico felix pods for metrics

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #527

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**: my 1st PR for apps please let me know if I missed something 

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [x] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/charts/calico-felix-metrics/templates/service.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
```
Comment (Xartos) [inline helmfile/charts/calico-felix-metrics/templates/service.yaml:1]: Don't we need anything else than adding this? Is the metrics enabled by default?

**Review by Xartos — COMMENTED** [review]
Some comments:

Please add a note to `WIP-CHANGELOG.md` also, `helmfile/values/calico-felix-metrics.yaml.gotmpl` and ` helmfile/charts/calico-felix-metrics/values.yaml` are both empty, do we still need them?

File: helmfile/charts/calico-felix-metrics/templates/service.yaml (hunk 1 above)
Comment (Xartos) [reply helmfile/charts/calico-felix-metrics/templates/service.yaml:1]: Just saw that this is enabled via kubespray :+1:

File: helmfile/charts/calico-felix-metrics/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# calico-felix-metrics
+
+This helm chart deploys `calico-felix-metrics`, an application that gathers metrics about NetworkPolicies.
```
[the remark this answers is not in the document]
Comment (crssnd) [reply helmfile/charts/calico-felix-metrics/README.md:3]: "Felix Dashboard (Calico)" dashboard shows the general health of Calico.
The existing "NetworkPolicy Dashboard" has "the more interesting" information, but it is handled by calico-accountant (different helm chart and app).

Makes sense to merge this 2 dashboards into one?

**Comment by crssnd** [comment]
> Some comments:
> 
> Please add a note to `WIP-CHANGELOG.md` also, `helmfile/values/calico-felix-metrics.yaml.gotmpl` and ` helmfile/charts/calico-felix-metrics/values.yaml` are both empty, do we still need them?

done

File: helmfile/charts/calico-felix-metrics/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Xartos) [reply helmfile/charts/calico-felix-metrics/README.md:3]: No, keep it separate. But I would then rephrase this to 
```suggestion
This helm chart deploys `calico-felix-metrics`, an application that gathers metrics about the general health of Calico.
```

File: helmfile/charts/calico-felix-metrics/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# calico-felix-metrics
+
+This helm chart deploys `calico-felix-metrics`, an application that gathers metrics about the general health of Calico.
+
```
Comment (Xartos) [inline helmfile/charts/calico-felix-metrics/README.md:4]: nit: Maybe add a sentence that this chart assumes you have setup felix to expose metrics. And also add a small snippet with a command to verify that it's enabled.
Just so that people assumes that it will work as soon as they have enabled this.

File: helmfile/50-applications.yaml [hunk 4]
```diff
@@ -96,6 +96,19 @@ releases:
...
+- name: calico-felix-metrics
+  namespace: kube-system
+  labels:
+    app: calico-felix-metrics
+  chart: ./charts/calico-felix-metrics
+  version: 0.1.0
+  installed: {{ .Values.calicoFelixMetrics.enabled }}
+  missingFileHandler: Error
+  values:
+  - values/calico-felix-metrics.yaml.gotmpl
```
Comment (viktor-f) [inline helmfile/50-applications.yaml:109]: This file does not exist, and it looks like it is not needed.
```suggestion
```

**Review by viktor-f — CHANGES_REQUESTED** [review]

File: helmfile/charts/calico-felix-metrics/README.md (hunk 3 above)
Comment (crssnd) [reply helmfile/charts/calico-felix-metrics/README.md:4]: done. please check

**Review by viktor-f — APPROVED** [review]
LGTM, just squash some commits.

**Review by Xartos — COMMENTED** [review]
Please add the pipeline config and then I'd be happy

**Comment by crssnd** [comment]
> Please add the pipeline config and then I'd be happy

I missed that. now should be ok. thanks!

**Review by Xartos — APPROVED** [review]
LGTM!

File: helmfile/charts/calico-felix-metrics/templates/service.yaml (hunk 1 above)
Comment (OlleLarsson) [inline helmfile/charts/calico-felix-metrics/templates/service.yaml:1]: Is the scrape_interval set to 30s by default?

Comment (crssnd) [reply helmfile/charts/calico-felix-metrics/templates/service.yaml:1]: yes
```
- job_name: serviceMonitor/kube-system/calico-felix-metrics/0
  honor_timestamps: true
  scrape_interval: 30s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  follow_redirects: true
  relabel_configs:
```
